src/data/event.py

```diff
@@ -916,8 +917,8 @@ def __lt__(self, other: 'Event'):
         # p1 < p2 calls p1.__lt__(p2)
         return self.uniq_id > other.uniq_id
 
-    def __eq__(self, other: 'Event'):
+    def __eq__(self, other: object) -> bool | NotImplementedType :
         # p1 == p2 calls p1.__eq__(p2)
-        if not isinstance(self, Event):
+        if not isinstance(other, Event):
```

Comment:
Good catch.
Also, for more generality:
```suggestion
        if not isinstance(other, self.__class__):
```